### ISSUE TITLE
[4.0] media action rotate plugin [a11y]

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#29943]
![image](https://user-images.githubusercontent.com/1296369/86500420-cbc35700-bd88-11ea-8290-b23b5a5cb727.png)

The 4 highlighted buttons only have a label of 0, 90 etc

The fieldset does have a label of "rotate_distinct" but that is just because no label is actually defined in the xml so it is also not translatable.

Because it is a button group it should be rendered as a legend not a label so that it describes the group of buttons.

**Comment by Quy** [comment]
Related #29821

**Comment by hans2103** [comment]
Test instructions:

- Go to Joomla Administrator > Content > Media
- Select an image
- Press the three dots to open a dropdown with options
- Press the pencil to edit the selected image
- Go to tab _rotate_
- See the screenshot above. 

## At this moment the screenshot below is what I see:
<img width="1336" alt="Schermafdruk 2020-07-22 22 41 08" src="https://user-images.githubusercontent.com/639822/88226607-9b2a5b00-cc6c-11ea-8071-087f41104399.png">

## The  corresponding HTML
<img width="670" alt="Schermafdruk 2020-07-22 22 43 48" src="https://user-images.githubusercontent.com/639822/88226741-d62c8e80-cc6c-11ea-8179-aa5d99d181a8.png">

## Suggestion
@brianteeman Would a hidden text improve a11y?
 `<span class="sr-only><?php echo Text::_('PLG_MEDIA-ACTION_ROTATE_PARAM_ANGLE_DEGREES'); ?></span>`

**Comment by brianteeman** [comment]
aria-label would be better for that one part of the issues

**Comment by hans2103** [comment]
This is what I was thinking about. 
```
<label for="jform_rotate_distinct0" class="btn btn-outline-secondary">
  <input type="radio" id="jform_rotate_distinct0" name="jform[rotate_distinct]" value="0">
  0 <span class="sr-only><?php echo Text::_('PLG_MEDIA-ACTION_ROTATE_PARAM_ANGLE_DEGREES'); ?></span>
</label>
```
```
PLG_MEDIA-ACTION_ROTATE_PARAM_ANGLE_DEGREES = "degrees"
```

Is this what you are thinking about?
```
<label for="jform_rotate_distinct0" class="btn btn-outline-secondary">
  <input type="radio" id="jform_rotate_distinct0" name="jform[rotate_distinct]" value="0" aria-label="<?php
  echo Text::sprintf('PLG_MEDIA-ACTION_ROTATE_PARAM_ANGLE_DEGREES', 0); ?>">
  0
</label>
```
```
PLG_MEDIA-ACTION_ROTATE_PARAM_ANGLE_DEGREES = "%s degrees"
```

**Comment by brianteeman** [comment]
TBH commenting on this one part of the problem is hard because of the fake button group etc

**Comment by hans2103** [comment]
We're improving it with small steps. Changing the fake button group into something else can be done later.
Adding aria-label is my focus at this moment. And by adding it we are improving the a11y of this part. 
Which of the two do you prefer?

**Comment by brianteeman** [comment]
Doing the aria-label the way you propose is pointless as when you get rid of the fake button group the labels will need redoing. 

You cant address the accessibility errors here without addressing them all at the same time. What you are proposing to do makes the label for attribute on the label useless and probably confusing to screen readers etc as you now have a label pointing to a label

**Comment by hans2103** [comment]
See the screenshots in my comment https://github.com/joomla/joomla-cms/issues/29943#issuecomment-662687680
The difference with the screenshot you provided is that there is a `label` above the numbers.
And the numbers are within a `label`.

Is changing the first `label` with text _Angle buttons_ into a `legend` element with the same text solving the a11y issue?
Otherwise... what needs to be done to change the current state to an a11y state?

**Comment by brianteeman** [comment]
The label above the numbers is an orphaned label (see explanation of what an orphaned label is below)

You cannot simply rename it as a legend. A legend must be inside the fieldset so it would need to be moved as well.

Orphaned form label
**What It Means**
A form label is present, but it is not correctly associated with a form control.
**Why It Matters**
An incorrectly associated label does not provide functionality or information about the form control to the user. It usually indicates a coding or other form labeling issues.
**How to Fix It**
Properly associate the label with its corresponding form control. If there is no corresponding form control, remove the label. Labels are not appropriate for image, submit, reset, button, or hidden form controls.

A `<label>` element:
does not surround a form control and the for attribute is missing/empty
references an element that is not present in the page
references an element that is not an <input>, <select> or <textarea> element
references an <input> element with image, submit, reset, button, or hidden type

**Comment by brianteeman** [comment]
... cont

A fieldset legend presents a description of the form elements within a fieldset. A legend should be provided if a higher level description is necessary for groups of check boxes, radio buttons, or other form controls.

If a higher level description is necessary for the user to understand the function or purpose of the controls within the fieldset, provide this description within the `<legend`>. 

If this description or grouping is not necessary, the fieldset should probably be removed. 

**Note that the legend is repeated to screen reader users for each form control within the fieldset.**

**Comment by dgrammatiko** [comment]
@brianteeman is this still valid?

**Comment by brianteeman** [comment]
yes - if anything it is worse now

**Comment by dgrammatiko** [comment]
It's weird as the layout that is used for the media plugins is the same as for other parts of the backend: https://github.com/joomla/joomla-cms/blob/a94decccac55949997c6fd0227b26abfc5457fec/administrator/components/com_media/tmpl/file/default.php#L69

**Comment by brianteeman** [comment]
what does that layout have to do with this

**Comment by dgrammatiko** [comment]
> what does that layout have to do with this

This is the file that renders the media action plugins

**Comment by brianteeman** [comment]
yes i know that obviously but that doesnt have anything to do with the error reported
